File: frontend/src/shared/api/employees.ts
import { axiosInstance } from '../config/axios';

export interface EmployeePhoto {
  photoWebp50?: string | null;
  photoWebp50sha256?: string;
  photoWebp300?: string | null;
  photoJpg300sha256?: string;
  photoWebp200?: string | null;
  photoWebp200Sha256?: string;
}

export interface Employee {
  hashMd5: string;
  fullName: string;
  employeePhoto?: EmployeePhoto | null;
  jobTitle?: string;
  [key: string]: unknown;
}

export interface EmployeesByHashesRequest {
  hashesMd5: string[];
  includePhoto?: boolean | null;
}

export const employeesApi = {
  searchByFio: async (searchFio: string, includePhoto: boolean = true): Promise<Employee[]> => {
    if (!searchFio || searchFio.length < 3) {
      return [];
    }

    const response = await axiosInstance.get<Employee[]>('/api/employees/search/', {
      params: { searchFio, includePhoto },
    });

    return response.data;
  },

  getByHash: async (hash: string, includePhoto: boolean = true): Promise<Employee | null> => {
    if (!hash) {
      return null;
    }

    const response = await axiosInstance.get<Employee>(`/api/employees/${hash}/`, {
      params: { includePhoto },
    });

    return response.data;
  },

  getByHashes: async (request: EmployeesByHashesRequest): Promise<Record<string, Employee>> => {
    if (!request.hashesMd5.length) {
      return {};
    }

    const response = await axiosInstance.post<Record<string, Employee>>(
      '/api/employees/by-hashes/',
      request
    );

    return response.data;
  },
};

File: backend/schemas/employees.py
from typing import Optional
from pydantic import BaseModel, ConfigDict


class EmployeesByHashesRequest(BaseModel):
    hashesMd5: list[str]
    includePhoto: Optional[bool] = False


class EmployeePhotoResponse(BaseModel):
    model_config = ConfigDict(extra="allow")

    photoWebp50: Optional[str] = None
    photoWebp50sha256: Optional[str] = None
    photoWebp300: Optional[str] = None
    photoJpg300sha256: Optional[str] = None
    photoWebp200: Optional[str] = None
    photoWebp200Sha256: Optional[str] = None


class EmployeeResponse(BaseModel):
    model_config = ConfigDict(extra="allow")

    hashMd5: str
    fullName: Optional[str] = None
    jobTitle: Optional[str] = None
    department: Optional[str] = None
    division: Optional[str] = None
    login: Optional[str] = None
    workingNowStatus: Optional[str] = None
    employeePhoto: Optional[EmployeePhotoResponse] = None
